src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -766,6 +766,47 @@ public long countSearchTransactions(final TransactionsSearchDto transactionsSear
 		}
 	}
 
+	public List<TransactionDto> getAvailableBalanceTransactions(final Integer lotId) {
+		try {
+
+			if (lotId != null) {
+				final StringBuilder sql = new StringBuilder(this.SEARCH_TRANSACTIONS_QUERY);
+				sql.append(" and (act.trnstat =").append(TransactionStatus.CONFIRMED.getIntValue()).append(" or (act.trnstat = ")
+					.append(TransactionStatus.PENDING.getIntValue()).
+					append(" and act.trntype = ").append(TransactionType.WITHDRAWAL.getId()).append(")) ");
+				sql.append(" and act.lotid = ").append(lotId).append(" ");
+				final SQLQuery query = this.getSession().createSQLQuery(sql.toString());
+				query.addScalar("transactionId");
+				query.addScalar("createdByUsername");
+				query.addScalar("transactionType");
+				query.addScalar("amount");
+				query.addScalar("notes");
+				query.addScalar("transactionDate", Hibernate.DATE);
+				query.addScalar("lotLotId");
+				query.addScalar("lotGid");
+				query.addScalar("lotDesignation");
+				query.addScalar("lotStockId");
+				query.addScalar("lotScaleId");
+				query.addScalar("lotScaleName");
+				query.addScalar("lotStatus");
+				query.addScalar("transactionStatus");
+				query.addScalar("lotLocationId");
+				query.addScalar("lotComments");
```

Comment:
@clarysabel can it be extracted to a method?
all SEARCH_TRANSACTIONS_QUERY queries will need to have the same scalars right?